versiondb/tsrocksdb/store.go

```diff
@@ -164,7 +166,9 @@ func (s Store) iteratorAtVersion(storeKey string, start, end []byte, version *in
 	prefix := storePrefix(storeKey)
 	start, end = iterateWithPrefix(prefix, start, end)
 
-	itr := s.db.NewIteratorCF(newTSReadOptions(version), s.cfHandle)
+	readOpts := newTSReadOptions(version)
+	defer readOpts.Destroy()
```

Comment:
not sure if it's referenced in the iterator itself.

Reply:
defer could hold the reference to avoid early release

Reply:
I mean the iterator is returned to the caller, and keep use after defer runs.

Reply:
iterator lifecycle is controlled by us, we only destroy readOpts after using

Reply:
Could long run test with this timestamp reference [change](https://github.com/linxGnu/grocksdb/pull/172), but I think grockdb expect us to defer a Destroy call which holds the other 4 references of bytes